src/main/java/com/algorand/algosdk/transaction/MethodCallParams.java

```diff
@@ -170,7 +170,7 @@ public List<TransactionWithSigner> createTransactions() {
                 methodABIts.add(argT.parsedType);
             } else
                 throw new IllegalArgumentException(
-                        "error: the type of method argument is a transaction-type, but no transaction-with-signer provided"
+                        "error: the type of method argument "+i+" is a "+argT.type+", but "+argT.type+" not provided"
```

Comment:
I think
```suggestion
                        "error: the type of method (" + this.method.name +") argument " + (i+1) + " is a " + this.methodArgs.get(i).getClass().getName() + ", but " + argT.type + " not provided"
```
`argT.type` on both side might be a typo? (or not as helpful)

I tested this and here's my suggestion's output

<img width="1635" alt="Screenshot 2023-03-01 at 9 58 24 PM" src="https://user-images.githubusercontent.com/834255/222344364-3a8e75ff-1025-4b29-a802-8ab898c86597.png">

and here is String as input

<img width="1707" alt="Screenshot 2023-03-01 at 9 56 05 PM" src="https://user-images.githubusercontent.com/834255/222344767-f4f9cec8-11c6-4807-9785-60061653c827.png">

Reply:
nice one, thanks!